### ISSUE TITLE
Refusing to deserialize untrusted content of type pickle

### ISSUE THREAD

**Issue by monegim** [danihodovic/celery-exporter#153]
```
File "/usr/local/lib/python3.9/site-packages/kombu/messaging.py", line 620, in _receive_callback
decoded = None if on_m else message.decode()
File "/usr/local/lib/python3.9/site-packages/kombu/message.py", line 194, in decode
self._decoded_cache = self._decode()
File "/usr/local/lib/python3.9/site-packages/kombu/message.py", line 198, in _decode
return loads(self.body, self.content_type,
File "/usr/local/lib/python3.9/site-packages/kombu/serialization.py", line 242, in loads
raise self._for_untrusted_content(content_type, 'untrusted')
kombu.exceptions.ContentDisallowed: Refusing to deserialize untrusted content of type pickle (application/x-python-serialize)
```
Deployed `celery-exporter` using helm chart.

**Comment by monegim** [comment]
> Could you post your helm config?

Do you mean `values.yaml`?

**Comment by monegim** [comment]
``` yaml

replicaCount: 1

image:
  repository: danihodovic/celery-exporter
  pullPolicy: IfNotPresent
  tag: ""

imagePullSecrets: []
nameOverride: ""
fullnameOverride: ""

serviceAccount:
  create: true
  annotations: {}
  name: ""

env: []

podAnnotations: {}

podSecurityContext: {}

securityContext: {}

service:
  type: ClusterIP
  port: 9808

ingress:
  enabled: false
  className: ""
  annotations: {}
  hosts:
    - host: celery-exporter.example
      paths:
        - path: /
          pathType: ImplementationSpecific
  tls: []

serviceMonitor:
  enabled: true
  additionalLabels: {}
  namespace: ""
  namespaceSelector: {}
  scrapeInterval: 30s
  targetLabels: []
  relabelings: []
  metricRelabelings: []

resources: 
  limits:
    cpu: 100m
    memory: 128Mi
  requests:
    cpu: 100m
    memory: 128Mi

nodeSelector: {}

tolerations: []

affinity: {}

```

**Comment by danihodovic** [comment]
cc @adinhodovic

**Comment by adinhodovic** [comment]
@monegim What Chart version are you using?

**Comment by monegim** [comment]
> @monegim What Chart version are you using?
```yaml
version: 0.4.1
appVersion: 0.5.3
```

**Comment by adinhodovic** [comment]
> > @monegim What Chart version are you using?
> 
> ```yaml
> version: 0.4.1
> appVersion: 0.5.3
> ```

https://docs.celeryq.dev/en/3.1/configuration.html

Try to see if you override the CELERY_ACCEPT_CONTENT environment variable. By default it should accept pickle content or any content. echo $CELERY_ACCEPT_CONTENT, or printenv

**Comment by monegim** [comment]
This has been set to this: 
https://github.com/DefectDojo/django-DefectDojo/blob/master/dojo/settings/settings.dist.py#L991

```python
CELERY_ACCEPT_CONTENT = ['pickle', 'json', 'msgpack', 'yaml']
```

**Comment by adinhodovic** [comment]
> This has been set to this: https://github.com/DefectDojo/django-DefectDojo/blob/master/dojo/settings/settings.dist.py#L991
> 
> ```python
> CELERY_ACCEPT_CONTENT = ['pickle', 'json', 'msgpack', 'yaml']
> ```

Try adding `application/x-python-serialize` to the list.

**Comment by monegim** [comment]
```python
CELERY_ACCEPT_CONTENT = ['pickle', 'json', 'msgpack', 'yaml', 'application/x-python-serialize']
```
This did not work. Is this issue related to the `celery-exporter` or `celery`?

**Comment by adinhodovic** [comment]
> ```python
> CELERY_ACCEPT_CONTENT = ['pickle', 'json', 'msgpack', 'yaml', 'application/x-python-serialize']
> ```
> 
> This did not work. Is this issue related to the `celery-exporter` or `celery`?

I believe celery does not accept the content celery-exporter wants.

**Comment by monegim** [comment]
So, should not something change in `celery-exporter`?

**Comment by danihodovic** [comment]
Feel free to submit a PR

**Comment by howardjones** [comment]
Hitting this too, while looking around for a prometheus option for celery. We use pickle encoding to allow us to pass dataclass objects to tasks (which the default json serializer can't handle). Does the exporter need to be given the same config object as the celery worker somehow? (with the `task_serializer` and `accept_content` set). My celery config object looks like this:

```
class CeleryConfig:
    task_serializer = 'pickle'
    result_serializer = 'pickle'
    event_serializer = 'pickle'
    result_accept_content = ['pickle']
    accept_content = ['application/json', 'application/x-python-serialize', 'pickle']
    worker_send_task_events = True
    task_send_sent_event = True
```

**Comment by howardjones** [comment]
(confirming - if you add your Celery config object in exporter.py, everything seems to work. Just need a way to do that without altering the code! :-) Not sure if everything we have can be done with just environment vars)

My ugly test case:
```
▶ git diff src/exporter.py                                    
diff --git a/src/exporter.py b/src/exporter.py
index b68a300..090096d 100644
--- a/src/exporter.py
+++ b/src/exporter.py
@@ -13,6 +13,15 @@ from prometheus_client import CollectorRegistry, Counter, Gauge, Histogram
 
 from .http_server import start_http_server
 
+class CeleryConfig:
+    task_serializer = 'pickle'
+    result_serializer = 'pickle'
+    event_serializer = 'json'
+    result_accept_content = ['pickle']
+    accept_content = ['application/json', 'application/x-python-serialize', 'pickle']
+    worker_send_task_events = True
+    task_send_sent_event = True
+
 
 class Exporter:  # pylint: disable=too-many-instance-attributes,too-many-branches
     state: State = None
@@ -199,6 +208,7 @@ class Exporter:  # pylint: disable=too-many-instance-attributes,too-many-branche
         logger.remove()
         logger.add(sys.stdout, level=click_params["log_level"])
         self.app = Celery(broker=click_params["broker_url"])
+        self.app.config_from_object(CeleryConfig)
         transport_options = {}
         for transport_option in click_params["broker_transport_option"]:
             if transport_option is not None:
```

**Comment by danihodovic** [comment]
Resolved by @amandahla in #186 . Run the container with `CE_ACCEPT_CONTENT="json,pickle"`
